include/sys/zfs_znode.h

```diff
@@ -38,7 +39,24 @@
 #include <sys/zfs_sa.h>
 #include <sys/zfs_stat.h>
 #include <sys/zfs_rlock.h>
+
+#else
+
+#include <sys/zfs_rlock.h>
+#include <sys/zfs_context.h>
+#include <sys/refcount.h>
+
+/* User mode znode emulation for ztest */
+#define ZTOZSB(zp) (zp)			/* unused */
+typedef struct znode {
+	uint64_t z_size;		/* unused, zeroed out */
+	uint64_t z_blksz;		/* unused, zeroed out */
+	uint64_t z_max_blksz;		/* unused, zeroed out */
+	zfs_rlock_t z_range_lock; 
```

Comment:
I think the only way to use zfs_rlock_t directly is to not declare a separate znode structure in user space and to assure the user mode always calls zfs_range_lock_impl() with zp = NULL. I could look into this, but is there a big difference ?